### PR TITLE
feat: support favorite migration also on departures response

### PR DESCRIPTION
**Pull request by marius-at-atb**
Migrate favorites using data received from the `/bff/v2/departures/departures` call

Depends on https://github.com/AtB-AS/atb-bff/pull/313

Resolves https://github.com/AtB-AS/kundevendt/issues/2732
(mentioned in this [comment](https://github.com/AtB-AS/kundevendt/issues/2732#issuecomment-1792286534))

### CODE REVIEW

File: src/place-screen/hooks/use-departures-data.ts [hunk 1]
```diff
@@ -337,6 +338,11 @@ export function useDeparturesData(
     !!state.tick,
   );
 
+  useEffect(() => {
+    const estimatedCalls = state.data;
+    estimatedCalls && potentiallyMigrateFavoriteDepartures(estimatedCalls);
+  }, [state.data, potentiallyMigrateFavoriteDepartures]);
```
Comment (gorandalum) [inline src/place-screen/hooks/use-departures-data.ts:344]: Is `potentiallyMigrateFavoriteDepartures` a stable reference?

**Review by gorandalum — APPROVED** [review]
Looks good!

File: src/place-screen/hooks/use-departures-data.ts (hunk 1 above)
Comment (marius-at-atb) [reply src/place-screen/hooks/use-departures-data.ts:344]: Good question!
Yes, both code walkthrough and testing indicates that it is stable.

Comment (marius-at-atb) [reply src/place-screen/hooks/use-departures-data.ts:344]: Aha 😅
I just looked at the current implementation and whether there was a way to trigger repeated potentiallyMigrateFavoriteDepartures changes from using the app. Will look into safeguarding it for the future as well :+1:

**Pull request merged**